[ONNX] Support restricted quantized range for activation.

BowenBao opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #76055
* #76002

PyTorch restricts activations to be in the range (0, 127).
In ONNX, the supported ranges are (0, 255) and (-128, 127),
respectfully, uint8 and int8. This PR extends support for range
(0, 127), by adding additional clipping when detected.

BowenBao committed:
[ONNX] Support restricted quantized range for activation.

PyTorch restricts activations to be in the range (0, 127).
In ONNX, the supported ranges are (0, 255) and (-128, 127),
respectfully, uint8 and int8. This PR extends support for range
(0, 127), by adding additional clipping when detected.

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Support restricted quantized range for activation."


PyTorch restricts activations to be in the range (0, 127).
In ONNX, the supported ranges are (0, 255) and (-128, 127),
respectfully, uint8 and int8. This PR extends support for range
(0, 127), by adding additional clipping when detected.

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Support restricted quantized range for activation."


PyTorch restricts activations to be in the range (0, 127).
In ONNX, the supported ranges are (0, 255) and (-128, 127),
respectfully, uint8 and int8. This PR extends support for range
(0, 127), by adding additional clipping when detected.

[ghstack-poisoned]

justinchuby commented on test/onnx/test_pytorch_onnx_onnxruntime.py:
```diff
@@ -8707,6 +8707,32 @@ def forward(self, input):
+    @disableScriptTest()  # RuntimeError: Can't redefine method: forward on class: __torch__.torch.nn.modules.linear.Linear
+    def test_fake_quantize_activation(self):
+        m = torch.nn.Linear(1, 1)
+        from torch import quantization
```
I would import everything first before setting up the rest of the test for clarity

BowenBao replied:
done

BowenBao committed:
Update on "[ONNX] Support restricted quantized range for activation."


PyTorch restricts activations to be in the range (0, 127).
In ONNX, the supported ranges are (0, 255) and (-128, 127),
respectfully, uint8 and int8. This PR extends support for range
(0, 127), by adding additional clipping when detected.

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Support restricted quantized range for activation."


PyTorch restricts activations to be in the range (0, 127).
In ONNX, the supported ranges are (0, 255) and (-128, 127),
respectfully, uint8 and int8. This PR extends support for range
(0, 127), by adding additional clipping when detected.

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Support restricted quantized range for activation."


PyTorch restricts activations to be in the range (0, 127).
In ONNX, the supported ranges are (0, 255) and (-128, 127),
respectfully, uint8 and int8. This PR extends support for range
(0, 127), by adding additional clipping when detected.

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Support restricted quantized range for activation."


PyTorch restricts activations to be in the range (0, 127).
In ONNX, the supported ranges are (0, 255) and (-128, 127),
respectfully, uint8 and int8. This PR extends support for range
(0, 127), by adding additional clipping when detected.

[ghstack-poisoned]

BowenBao commented:
@pytorchbot merge this
